src/Lean/Server/FileWorker/RequestHandling.lean

```diff
@@ -151,12 +151,21 @@ def locationLinksOfInfo (kind : GoToKind) (ictx : InfoWithCtx)
   let i := ictx.info
   let ci := ictx.ctx
   let children := ictx.children
-  match i with
-  | .ofTermInfo ti =>
+
```

Comment:
@mhuisi Can I ask you to take a look at my changes here? I refactored it a bit, added some `consumeMData`s that avoid some issues, but also add in specific handling for `sorry`, which is awkward.

I'm wondering it would be better if TermInfo/OmissionInfo would include an optional source position field. Maybe this could be deferred until we finally make a separate TermInfo-like type for delaboration?